jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -271,15 +278,22 @@ public boolean removeJob(JobHandle jobHandle, Timer ejbTimer) {
                 try {
                     ((TimerHandle) ejbTimerHandle).getTimer().cancel();
                 } catch (Throwable e) {
-                    logger.debug("Timer cancel error due to {}", e.getMessage());
+                    logger.warn("Timer cancel error for handle {}", ejbTimerHandle,  e);
                     return false;
                 }
                 return true;
+            } else {
+                logger.warn("No TimerHandle found for {}: {}", ejbHandle, ejbTimerHandle);
             }
         } else {
             logger.warn("No timerJobInstance available for {}", ejbHandle);
         }
 
+        if (!searchIfFailed) {
```

Comment:
Is this change really needed? ( I have not gone through latest traces yet), but I feel this will prevent searching in case the search migh be needed ( to be honest, I do not completely understand the need for the search, so I was being conservative in the first commit and only disable it when it is clear that the times is there and the cancel has failed)

Reply:
After reviewing the logs, this change makes sense. Probably we need more...